tests/Sharpliner.Tests/AzureDevOps/Pipelines/XHarnessPipeline.cs

```diff
@@ -1,315 +1,276 @@
-                })
+    public override string TargetFile => "eng/common/templates-official/post-build/post-build.yml";
+    public override TargetPathType TargetPathType => TargetPathType.RelativeToCurrentDir;
+    public override ConditionedList<JobBase> Definition =>
+    [
+    ];
+}
 
-                .StageTemplate("eng/e2e-test.yml", new()
-                {
-                    { "name", "E2E_Apple_Simulator_Mgmt" },
-                    { "displayName", "Apple - Simulator management" },
-                    { "testProject", "$(Build.SourcesDirectory)/tests/integration-tests/Apple/SimulatorInstaller.Tests.proj" },
-                })
+public class CommonPostBuildParameters : CorePostBuildParametersBase<CommonPostBuildParameters>
+{
+    public override bool Is1ESPipeline { get; set; } = false;
+}
 
-                .StageTemplate("eng/e2e-test.yml", new()
-                {
-                    { "name", "E2E_WASM" },
-                    { "displayName", "WASM" },
-                    { "testProject", "$(Build.SourcesDirectory)/tests/integration-tests/WASM/WASM.Helix.SDK.Tests.proj" },
-                }),
-
-            // NuGet publishing
-            If.Equal(variables["_RunAsInternal"], "True")
-                .StageTemplate("eng/common/templates/post-build/post-build.yml", new()
+public partial class CorePostBuild : StageTemplateDefinition<CorePostBuildParameters>
+{
+    public override string TargetFile => "eng/common/core-templates/post-build/post-build.yml";
+    public override TargetPathType TargetPathType => TargetPathType.RelativeToCurrentDir;
+    public override ConditionedList<Stage> Definition =>
+    [
+        If.Or(
+                Equal(parameters.For(x => x.EnableNugetValidation), "true"),
+                Equal(parameters.For(x => x.EnableSigningValidation), "true"),
+                Equal(parameters.For(x => x.EnableSourceLinkValidation), "true"),
+                Equal(parameters.For(x => x.SDLValidationParameters.Enable), "true")
+            ).Stage(new("Validate", "Validate Build Assets")
+            {
+                DependsOn = [parameters.For(x => x.ValidateDependsOn)],
+                Variables =
                 {
-                    { "publishingInfraVersion", 3 },
-                    { "enableSymbolValidation", true },
-                    { "enableSourceLinkValidation", true },
-                    { "validateDependsOn", new[] { "Build_Windows_NT" } },
-                    { "publishDependsOn", new[] { "Validate" } },
-
-                    // This is to enable SDL runs part of Post-Build Validation Stage
-                    { "SDLValidationParameters", new TemplateParameters()
+                    Template("/eng/common/core-templates/post-build/common-variables.yml"),
+                    new VariableTemplate("/eng/common/core-templates/variables/pool-providers.yml")
+                    {
+                        Parameters = new()
```

Comment:
this could work for parameters, having a `.For()` method on the `parameters` that gets passed a lambda-expression, this will not require any source-generation

The idea that the base `StageTemplateDefinition<T>` class overrides the `parameters` to include this new method

@premun thoughts?